backend/application/syson-application-configuration/src/main/java/org/eclipse/syson/application/libraries/SysONLibraryLoadingUtils.java

```diff
@@ -0,0 +1,152 @@
+     * @param definition
+     *            the (non-{@code null}) {@link SysONLibraryLoadingDefinition}.
+     * @return a (non-{@code null}) {@link URI} that uniquely identifies {@code resource}.
+     */
+    public static URI createEmfUriFor(final ClassPathResource resource, final SysONLibraryLoadingDefinition definition) {
+        final URI uri = URI.createURI(definition.scheme() + ":///" + UUID.nameUUIDFromBytes(resource.getPath().getBytes()));
+        return uri;
+    }
+
+    /**
+     * Creates the EMF {@link Resource} corresponding to a library {@link org.springframework.core.io.Resource} from the
+     * classpath, loads its contents and adds it to the given {@link ResourceSet}.
+     *
+     * @param springResource
+     *            the {@link org.springframework.core.io.Resource} to load.
+     * @param uri
+     *            the (non-{@code null}) {@link URI} to use for the created {@link Resource}.
+     * @param resourceSet
+     *            the (non-{@code null}) {@link ResourceSet} to load the created {@link Resource} into.
+     * @param resourceLoadingBehavior
+     *            the (non-{@code null}) behavior for creating a {@link Resource} from {@code uri}.
+     * @return the (non-{@code null}) newly-created {@link Resource}.
+     * @throws SysONLibraryLoadingException
+     */
+    public static Resource createAndLoadEmfResourceFor(final org.springframework.core.io.Resource springResource,
+            final URI uri, final ResourceSet resourceSet, final Function<URI, Resource> resourceLoadingBehavior) throws SysONLibraryLoadingException {
+        final Resource emfResource = resourceLoadingBehavior.apply(uri);
+        try (InputStream inputStream = new ByteArrayInputStream(springResource.getContentAsByteArray())) {
+            resourceSet.getResources().add(emfResource);
+            emfResource.load(inputStream, Map.of());
+            emfResource.eAdapters().add(new ResourceMetadataAdapter(FilenameUtils.getBaseName(springResource.getFilename())));
+            return emfResource;
+        } catch (IOException ioException) {
+            resourceSet.getResources().remove(emfResource);
+            throw new SysONLibraryLoadingException("There was an unexpected issue while loading the contents of '%s'".formatted(springResource.getFilename()),
+                    ioException);
+        }
+    }
+
+    /**
+     * Explores the classpath of the application to find all resources corresponding to a library.
+     *
+     * @param definition
+     *            the (non-{@code null}) {@link SysONLibraryLoadingDefinition}.
+     * @return a (non-{@code null}) {@link List} of (non-{@code null}) {@link ClassPathResource}.
+     * @throws SysONLibraryLoadingException
+     */
+    public static List<ClassPathResource> findAllLibraryResourcesFromApplicationClasspath(final SysONLibraryLoadingDefinition definition)
+            throws SysONLibraryLoadingException {
```

Comment:
Why the new line for the throws SysONLibraryLoadingException?
It seems not necessary